app/services/user.js

```diff
@@ -0,0 +1,58 @@
+const userRepository = require('../repositories/user');
+const error = require('../services/error');
+
+class UserService {
+  async getUser(id) {
+    const foundUser = await userRepository.getUser(id);
+    if (!foundUser || foundUser.deletedAt) {
+      error.throwNotFoundError("User not found.");
+    }
+    return foundUser;
+  }
+
+  async createUser(user) {
+    let userExists;
+    try {
+      userExists = await userRepository.getUserByEmail(user.email);
+    } catch (err) {
+      error.throwValidationError("Email is required.");
+    }
+    if (!userExists) {
+      try {
+        return await userRepository.createUser(user);
+      } catch (err) {
+        error.throwValidationError("User data invalid.");
+      }
+    } else {
+      error.throwValidationError("Email already used.");
+    }
+  }
+
+  async deleteUser(id) {
+    await this.getUser(id);
+    await userRepository.deleteUser(id);
+    return await userRepository.getUserNonParanoid(id);
+  }
+
+  async updateUser(id, newUserData) {
+    await this.getUser(id);
```

Comment:
Here as well you should throw an error if the user doesn't exist. At this point a light bulb should turn on (actually in the previous function). This is the 3rd time when you want to throw an error if you don't have a user and instead of writing the same code 3 times it is better to have a function that does this.